shell/platform/android/io/flutter/view/TextureRegistry.java

```diff
@@ -39,4 +40,37 @@
          */
         void release();
     }
+  
+   /**
+   * inform flutter engine that an new frame is render,  start an  new render pipeline
+   */
+    void onShareFrameAvaliable(int textureIndex);
+  
+   /**
+   * get flutter's opengl context
+   */
+    EGLContext getShareContext();
+  
+   /**
+   * Creates a Share Texture which use a texture create by native
+   *
+   * @return A SurfaceTextureEntry.
+   */
+    ShareTextureEntry createShareTexture(long shareTextureID);
+  
+    /**
+     * A registry entry for a managed ShareTexture.
+     */
+    interface ShareTextureEntry {
+      
+        /**
+         * @return The identity of this ShareTexture.
+         */
+        long id();
+      
+        /**
+         * Deregisters and releases this ShareTexture.
```

Comment:
All of our logic about the ShareTextureEntry's life circle  is  the same as the original external texture entry namely SurfaceTextureEntry。
When called multiple times, it will execute once。
And when the object created,  it can be called.